rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <VX/vx.h>
+#include <VX/vx_types.h>
+#include <vector>
+#include <cstring>
+#include <array>
+#include <queue>
+#include <memory>
+#if ENABLE_HIP
+#include "hip/hip_runtime.h"
+#include "device_manager_hip.h"
+#else
+#include "device_manager.h"
+#endif
+#include "commons.h"
+
+/*! \brief Converts Rocal Memory type to OpenVX memory type
+ *
+ * @param mem input Rocal type
+ * @return the OpenVX type associated with input argument
+ */
+vx_enum vx_mem_type(RocalMemType mem);
+
+vx_size tensor_data_size(RocalTensorDataType data_type);
+
+/*! \brief Holds the information about a rocALTensor */
+class rocALTensorInfo
```

Comment:
We have image (which is named Image class). If we need to add rocal before tensor, probably we need to add before image as well. Also I prefer rocalTensor. Please don't use rocAL for naming variables inside the implementation

Reply:
We will change the name of the class to rocalTensor